cmake/thirdparty/SetupMPI.cmake

```diff
@@ -123,16 +146,33 @@ if (ENABLE_FIND_MPI)
 endif()
 
 # Allow users to override CMake's FindMPI
+# Assuming we are not directly using targets it provides
 if (BLT_MPI_COMPILE_FLAGS)
+    if(BLT_USE_FIND_MPI_TARGETS)
```

Comment:
Can you please convert this to a `foreach()` ?
That way it is easier to see that it is the same logic.

Reply:
Each error case here provides a detailed unique error message. It's not just the var names that are different, so var indirection in a for loop won't get you the same result.

Reply:
Looking at the full section in the file -- the logic seems easy to parse.  I feel a cmake for loop would not improve the readability. If you have an idea of how it could work, lets explore in a future PR.